src/lib.rs

```diff
@@ -1,6 +1,7 @@
 //#![warn(missing_doc)]
 #![forbid(non_camel_case_types)]
-#![feature(slicing_syntax)]
```

Comment:
Slicing_syntax has been added into the language.
However I replaced `[]` by `as_slice` because `[]` gave cryptic error messages. They reworked the way this works, so I'm not sure about what is happening.